spec/models/custom_field/calculated_value_spec.rb

```diff
@@ -56,13 +149,193 @@
     end
 
     it "returns the formula as a string" do
-      formula = "1 * cf_7 + cf_42"
+      formula = "1 * {{cf_7}} + {{cf_42}}"
       subject.formula = formula
 
       expect(subject.formula_string).to eq(formula)
     end
   end
 
+  describe "#formula_referenced_custom_field_ids" do
+    it "returns an empty array if no formula is set" do
+      subject.formula = nil
+
+      expect(subject.formula_referenced_custom_field_ids).to eq([])
+    end
+
+    it "returns an empty array if formula doesn't reference custom fields" do
+      subject.formula = "2 + 2"
+
+      expect(subject.formula_referenced_custom_field_ids).to eq([])
+    end
+
+    it "returns ids if formula references custom fields" do
+      subject.formula = "1 * {{cf_7}} + {{cf_42}}"
+
+      expect(subject.formula_referenced_custom_field_ids).to eq([7, 42])
+    end
+  end
+
+  describe "#formula_contains_reference_to_id?" do
+    let!(:int_field) { create(:project_custom_field, :integer, default_value: 10, is_for_all: true) }
+    let!(:float_field) { create(:project_custom_field, :float, default_value: 5.5, is_for_all: true) }
+    let!(:text_field) { create(:project_custom_field, :text, default_value: "text", is_for_all: true) }
+
+    current_user { create(:admin) }
+
+    context "when checking a non-calculated value custom field" do
+      it "returns false for integer custom field" do
+        expect(int_field.formula_references_id?(subject.id)).to be false
```

Comment:
Just an idea, really not sure if it is better:
```suggestion
        expect(int_field).to be_formula_references_id(subject.id)
```

Reply:
I thought about that, but with the method name being what it is, it does not read like proper English and feels off.